src/Bicep.Core/Analyzers/Linter/Rules/BCPL1040.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class BCPL1040 : LinterRuleBase
```

Comment:
The rule is more complicated than is implemented here, see https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/test-cases#location-uses-parameter:
> Your templates should have a parameter named location. Use this parameter for setting the location of resources in your template. In the main template (named azuredeploy.json or mainTemplate.json), this parameter can default to the resource group location. In linked or nested templates, the location parameter shouldn't have a default location.

Reply:
Removed for a separate PR